### PR TITLE
Removes lib-auth and setup lib-classifier with PJC auth

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-auth, lib-classifier, app-project

Describe your changes:
This removes lib-auth, updates the bootstrap bash script, updates lib-classifier to use PJC's auth client (or oauth client), and updates app-project to remove the workaround for the mixed use of the old and new clients. 

To test it out, the dev version of the classifier: `npm run dev`
Log in and out. You'll see in the mobx dev tools a UPP is set when you sign in and reset when you sign out. 

I included one minor style fix with the footer and theme style updates that got merged today.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.js [hunk 1]
```diff
@@ -64,10 +48,29 @@ const UserProjectPreferencesStore = types
...
+
+      try {
+        const bearerToken = yield getBearerToken(authClient)
+        const user = yield authClient.checkCurrent()
+        if (bearerToken && user) {
+          self.fetchUPP(bearerToken, user)
+        } else {
+          self.reset()
+        }
+      } catch (e) {
```
Comment (rogerhutchings) [inline packages/lib-classifier/src/store/UserProjectPreferencesStore.js]: Should be `error`

**Comment by srallen** [comment]
I'd like to add a test for the helper function for getting the bearer token and there are broken tests related to the metadata modal (my bad for not checking in the recent PR), but I'd like to do that in a separate PR.

**Comment by eatyourgreens** [comment]
I'm seeing MST failing to resolve references when loading the classifier on http://localhost.zooniverse.org:3000/projects/eatyourgreens/-project-testing-ground/classify I assume that's #444?

http://localhost.zooniverse.org:3000/projects/eatyourgreens/-project-testing-ground/classify?env=staging crashes the React app with an invariant violation.

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.js [hunk 2]
```diff
@@ -64,10 +48,29 @@ const UserProjectPreferencesStore = types
...
     }
 
-    function * fetchUPP (bearerToken, project, user) {
+    function * checkForUser () {
+      const { authClient } = getRoot(self)
+
+      try {
+        const bearerToken = yield getBearerToken(authClient)
+        const user = yield authClient.checkCurrent()
+        if (bearerToken && user) {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/UserProjectPreferencesStore.js]: If there's a user, there must be a valid bearer token, so checking both is redundant here?

File: packages/lib-classifier/src/store/utils/getBearerToken.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 export default async function getBearerToken (authClient) {
   if (authClient) {
-    const authToken = await authClient.getBearerToken() || {}
-    if (authToken && authToken.token) {
-      return `Bearer ${authToken.token}`
+    const authToken = await authClient.checkBearerToken() || ''
+    if (authToken && authToken.access_token) {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/utils/getBearerToken.js]: I think `auth.checkBearerToken()` yields a string, not an object.

Comment (srallen) [reply packages/lib-classifier/src/store/utils/getBearerToken.js]: That's what I thought too, but it is returning an object. At least the oauth client does. I'll check the behavior of the first party auth client. It was a mistake for me to change the default be string if it's returning an object.

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.js (hunk 2 above)
Comment (srallen) [reply packages/lib-classifier/src/store/UserProjectPreferencesStore.js]: We need the user id for the query param and a defined bearer token for making the request authenticated. It does seem redundant, but I'm not totally trusting of the PJC auth client. I don't want to make the request if for some reason either of those are undefined.

File: packages/lib-classifier/src/store/utils/getBearerToken.js (hunk 3 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/store/utils/getBearerToken.js]: That's confusing. `auth.checkBearerToken()` yields a string for me in #437 (and is used that way earlier in this PR too.)
https://github.com/zooniverse/front-end-monorepo/blob/7ef5dd36d4be46b44c102a4aa59a55f3efc3faf9/packages/app-project/stores/Collections.js#L62-L63

Comment (srallen) [reply packages/lib-classifier/src/store/utils/getBearerToken.js]: Yep, I'm thinking it's a difference between auth and oauth cilents. Even though they have the same API method, it's returning different things.

Comment (srallen) [reply packages/lib-classifier/src/store/utils/getBearerToken.js]: I've confirmed that the auth client (first party, being used by app-project) returns a string of just the bearer token. The same method for the oauth client returns an object which includes the bearer token and expiration and the token type. So, given that the classifier is meant to be consumable by either, I'm going to have to write the helper to handle either for now.

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/utils/getBearerToken.js]: I remember now. First party handles token refresh for you, and always returns a valid token. Third party expects you to manage expired tokens, so returns an object including the expiration info.

**Comment by srallen** [comment]
@eatyourgreens I've refactored the helper function in the classifier to be able to deal with both the first party auth and oauth clients from PJC. I added specs to make sure it's doing what we think it is.

Could you manually test with the mini dev app in the lib-classifier instead of app-project for now (also just `npm run dev` in its folder)? With the app-project, the user store isn't written to manage or track the state of the request for the user, so what's happening if you test with a project that isn't crashing is that the classifier stores initialize, including the one for the UPP, before the request for the user finishes. We should refactor this so the store is handling the async requests and tracking the state of that request. The classifier should not load until we know there is a user or not. However, that refactor is out of scope for this PR.

As for your test project, I think it is crashing because the default workflow has audio/image subjects and we haven't built the new classifier to deal with those or have a reasonable error UI yet.

**Comment by eatyourgreens** [comment]
@srallen sounds good. I've been testing #437 with my test project, though, and not seeing errors there.

**Pull request merged**